### PR TITLE
Feature/event type filtering

### PR DESCRIPTION
**Pull request by ganfra**
This PR changes filtering for RoomSummary unread/preview and allow users to show/hide room member state events. 
This is a simple one waiting for filtering to be fully reworked.

Related to theses issues: 
 - Rooms incorrectly marked as unread (#588)
 - Option to hide joins/parts (#1231)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryUpdater.kt [hunk 1]
```diff
@@ -61,28 +60,6 @@ internal class RoomSummaryUpdater @Inject constructor(
         private val timelineEventDecryptor: Lazy<TimelineEventDecryptor>,
         private val eventBus: EventBus) {
 
-    companion object {
-        // TODO: maybe allow user of SDK to give that list
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryUpdater.kt:65]: One TODO done!

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import im.vector.app.core.resources.UserPreferencesProvider
+import org.matrix.android.sdk.api.session.events.model.EventType
+import org.matrix.android.sdk.api.session.room.timeline.TimelineEventFilters
+import org.matrix.android.sdk.api.session.room.timeline.TimelineSettings
+import javax.inject.Inject
+
+class TimelineSettingsFactory @Inject constructor(private val userPreferencesProvider: UserPreferencesProvider) {
+
+    fun create(): TimelineSettings {
+        return if (userPreferencesProvider.shouldShowHiddenEvents()) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt:28]: Just a remark, the code is (still) hard to understand, I already try to remove this if, but was never confident enough to commit it

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

**Comment by vinzv** [comment]
Hooray, thanks a lot for implementing this! :partying_face:

**Pull request merged**